File: fontes/frontend/Flutter/provider/fenix/lib/src/model/cadastros/banco_agencia.dart
import 'dart:convert';

import 'package:fenix/src/model/cadastros/banco.dart';

class BancoAgencia {
  int id;
  int idBanco;
  String numero;
  String digito;
  String nome;
  String telefone;
  String contato;
  String observacao;
  String gerente;
  Banco banco;

  BancoAgencia(
      {this.id, this.idBanco, this.numero, this.digito, this.nome, this.telefone, this.contato, this.observacao, this.gerente, this.banco});

  static List<String> campos = <String>['id', 'id_banco', 'numero', 'digito', 'nome', 'telefone', 'contato', 'observacao', 'gerente'];
  static List<String> colunas = <String>['id', 'Id Banco', 'Número', 'Dígito', 'Nome', 'Telefone', 'Contato', 'Observação', 'Gerente'];

  BancoAgencia.fromJson(Map<String, dynamic> jsonDados) {
    id = jsonDados['id'];
    idBanco = jsonDados['idBanco'];
    numero = jsonDados['numero'];
    digito = jsonDados['digito'];
    nome = jsonDados['nome'];
    telefone = jsonDados['telefone'];
    contato = jsonDados['contato'];
    observacao = jsonDados['observacao'];
    gerente = jsonDados['gerente'];
    banco = new Banco.fromJson(jsonDados['banco']);
  }

  Map<String, dynamic> get toJson {
    Map<String, dynamic> jsonDados = Map<String, dynamic>();
    jsonDados['id'] = this.id;
    jsonDados['idBanco'] = this.idBanco;
    jsonDados['numero'] = this.numero;
    jsonDados['digito'] = this.digito;
    jsonDados['nome'] = this.nome;
    jsonDados['telefone'] = this.telefone;
    jsonDados['contato'] = this.contato;
    jsonDados['observacao'] = this.observacao;
    jsonDados['gerente'] = this.gerente;
    jsonDados['banco'] = this.banco.toJson;
    return jsonDados;
  }
}

String bancoAgenciaEncodeJson(BancoAgencia bancoAgencia) {
  final jsonDados = bancoAgencia.toJson;
  return json.encode(jsonDados);
}

File: fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/model/cadastros/BancoAgencia.java
package com.t2ti.fenix.model.cadastros;

import java.io.Serializable;
import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.util.List;


/**
 * The persistent class for the banco_agencia database table.
 * 
 */
@Entity
@Table(name="banco_agencia")
@NamedQuery(name="BancoAgencia.findAll", query="SELECT b FROM BancoAgencia b")
public class BancoAgencia implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;

	private String numero;
	
	private String digito;

	private String nome;

	private String telefone;
	
	private String contato;
	
	private String observacao;
	
	private String gerente;


	//bi-directional many-to-one association to BancoAgencia
	//@JsonIgnore
	@ManyToOne
	@JoinColumn(name="ID_BANCO")
	private Banco banco;

	//bi-directional many-to-one association to BancoContaCaixa
	//@OneToMany //(mappedBy="bancoAgencia", cascade = CascadeType.ALL, orphanRemoval = true)
	//private List<BancoContaCaixa> bancoContaCaixas;

	public BancoAgencia() {
	}

	public int getId() {
		return this.id;
	}

	public String getContato() {
		return this.contato;
	}

	public void setContato(String contato) {
		this.contato = contato;
	}

	public String getDigito() {
		return this.digito;
	}

	public void setDigito(String digito) {
		this.digito = digito;
	}

	public String getGerente() {
		return this.gerente;
	}

	public void setGerente(String gerente) {
		this.gerente = gerente;
	}

	public String getNome() {
		return this.nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getNumero() {
		return this.numero;
	}

	public void setNumero(String numero) {
		this.numero = numero;
	}

	public String getObservacao() {
		return this.observacao;
	}

	public void setObservacao(String observacao) {
		this.observacao = observacao;
	}

	public String getTelefone() {
		return this.telefone;
	}

	public void setTelefone(String telefone) {
		this.telefone = telefone;
	}

	public Banco getBanco() {
		return this.banco;
	}

	public void setBanco(Banco banco) {
		this.banco = banco;
	}

//	public List<BancoContaCaixa> getBancoContaCaixas() {
//		return this.bancoContaCaixas;
//	}
//
//	public void setBancoContaCaixas(List<BancoContaCaixa> bancoContaCaixas) {
//		this.bancoContaCaixas = bancoContaCaixas;
//		for(BancoContaCaixa bancoContaCaixa : this.bancoContaCaixas) {
//			bancoContaCaixa.setBancoAgencia(this);
//		}
//	}

//	public BancoContaCaixa addBancoContaCaixa(BancoContaCaixa bancoContaCaixa) {
//		getBancoContaCaixas().add(bancoContaCaixa);
//		bancoContaCaixa.setBancoAgencia(this);
//
//		return bancoContaCaixa;
//	}
//
//	public BancoContaCaixa removeBancoContaCaixa(BancoContaCaixa bancoContaCaixa) {
//		getBancoContaCaixas().remove(bancoContaCaixa);
//		bancoContaCaixa.setBancoAgencia(null);
//
//		return bancoContaCaixa;
//	}

}